internal/telemetry/telemetry.go

```diff
@@ -0,0 +1,229 @@
+	Disabled          bool
+	BatchSize         int
+	BatchInterval     int
+	HookdeckSourceURL string
+	SentryDSN         string
+}
+
+func New(logger *logging.Logger, config TelemetryConfig) Telemetry {
+	if config.Disabled {
+		return &NoopTelemetry{}
+	}
+	return &telemetryImpl{
+		logger: logger,
+		config: config,
+	}
+}
+
+type telemetryImpl struct {
+	logger    *logging.Logger
+	config    TelemetryConfig
+	eventChan chan telemetryEvent
+	done      chan struct{}
+	client    *http.Client
+}
+
+func (t *telemetryImpl) sendEvent(event telemetryEvent) {
+	select {
+	case t.eventChan <- event:
+	default:
+		t.logger.Warn("telemetry event channel is full, dropping event", zap.Any("event", event))
+	}
+}
+
+func (t *telemetryImpl) processEvents() {
+	ticker := time.NewTicker(time.Duration(t.config.BatchInterval) * time.Second)
+	defer ticker.Stop()
+
+	batch := make([]telemetryEvent, 0)
+
+	for {
+		select {
+		case event := <-t.eventChan:
+			batch = append(batch, event)
+			if len(batch) >= t.config.BatchSize {
+				t.sendBatch(batch)
+				batch = batch[:0]
+			}
+		case <-ticker.C:
+			if len(batch) > 0 {
+				t.sendBatch(batch)
+				batch = batch[:0]
+			}
+		case <-t.done:
+			close(t.eventChan) // Stop accepting new events
+			for event := range t.eventChan {
+				batch = append(batch, event)
+			}
+			if len(batch) > 0 {
+				t.sendBatch(batch)
+			}
+			return
+		}
+	}
+}
+
+func (t *telemetryImpl) sendBatch(batch []telemetryEvent) {
+	t.logger.Debug("sending telemetry batch", zap.Int("size", len(batch)))
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	// Create worker pool
+	maxWorkers := min(len(batch), 10)
+	jobs := make(chan telemetryEvent, len(batch))
+	var wg sync.WaitGroup
+
+	// Start workers
+	for i := 0; i < maxWorkers; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for event := range jobs {
+				body, err := json.Marshal(event)
+				if err != nil {
+					t.logger.Debug("failed to marshal event", zap.Error(err))
+					continue
+				}
+
+				req, err := http.NewRequestWithContext(ctx, "POST", t.config.HookdeckSourceURL, bytes.NewBuffer(body))
+				if err != nil {
+					t.logger.Debug("failed to create request", zap.Error(err))
+					continue
+				}
+				req.Header.Set("Content-Type", "application/json")
+
+				resp, err := t.client.Do(req)
+				if err != nil {
+					t.logger.Debug("failed to send event", zap.Error(err))
+					continue
+				}
+				if resp.StatusCode >= 400 {
+					t.logger.Debug("failed to send event",
+						zap.Int("status", resp.StatusCode),
+						zap.Any("event", event))
+				}
+				resp.Body.Close()
+			}
+		}()
+	}
+
+	// Send jobs to workers
+	for _, event := range batch {
+		jobs <- event
+	}
+	close(jobs)
+
+	// Wait for all workers
+	wg.Wait()
+}
+
+func (t *telemetryImpl) Init(ctx context.Context) {
+	if err := sentry.Init(sentry.ClientOptions{
+		Dsn:              t.config.SentryDSN,
+		EnableTracing:    true,
+		TracesSampleRate: 1, // FIXME
```

Comment:
Which value should we use here?

Reply:
@alexluong - I'm not sure what you're looking for. Can you provide more info/more context?